src/graphene.cpp

```diff
@@ -475,49 +475,45 @@ bool CGrapheneBlock::process(CNode *pfrom,
         for (auto &kv : orphanpool.mapOrphanTransactions)
         {
             uint64_t cheapHash = kv.first.GetCheapHash();
-
-            if (mapPartialTxHash.count(cheapHash)) // Check for collisions
-                collision = true;
-
-            mapPartialTxHash[cheapHash] = kv.first;
+            auto ir = mapPartialTxHash.insert(std::make_pair(cheapHash, kv.first));
+            if (!ir.second)
+                collision = true; // insert returns false if no insertion
         }
 
         // We don't have to keep the lock on mempool.cs here to do mempool.queryHashes
         // but we take the lock anyway so we don't have to re-lock again later.
         LOCK(cs_xval);
-        mempool.queryHashes(memPoolHashes);
-
-        for (const uint256 &hash : memPoolHashes)
+        if (!collision)
         {
-            uint64_t cheapHash = hash.GetCheapHash();
+            mempool.queryHashes(memPoolHashes);
```

Comment:
why the extra parentheses here?

Reply:
I want to skip both the mempool.queryHashes(memPoolHashes) and the for loop if there is already a collision

Reply:
yup ok, i didn't see the !collision...